rocAL/rocAL/source/api/rocal_api_augmentation.cpp

```diff
@@ -517,19 +541,69 @@ rocalResize(
     auto input = static_cast<Image*>(p_input);
     try
     {
-        // For the resize node, user can create an image with a different width and height
+        if((dest_width || dest_height || resize_longer || resize_shorter) == 0)
+            THROW("Atleast one size 'dest_width' or 'dest_height' or 'resize_shorter' or 'resize_longer' must be specified")
+        if((dest_width || dest_height) && (resize_longer || resize_shorter))
+            THROW("Only one method of specifying size can be used \ndest_width and/or dest_height\nresize_shorter\nresize_longer")
+        if(resize_longer && resize_shorter)
+            THROW("'resize_longer' and 'resize_shorter' cannot be passed together. They are mutually exclusive.")
+
         ImageInfo output_info = input->info();
-        if (dest_width == 0) dest_width = input->info().width();
-        if (dest_height == 0) dest_height = input->info().height_single();
+        unsigned dst_width, dst_height;
+        RocalResizeScalingMode resize_scaling_mode;
+
+        // Change the scaling mode if resize_shorter or resize_longer is specified
+        if(resize_shorter) {
+            resize_scaling_mode = RocalResizeScalingMode::ROCAL_SCALING_MODE_NOT_SMALLER;
+            dst_width = dst_height = resize_shorter;
+        } else if(resize_longer) {
+            resize_scaling_mode = RocalResizeScalingMode::ROCAL_SCALING_MODE_NOT_LARGER;
+            dst_width = dst_height = resize_longer;
+        } else {
+            resize_scaling_mode = scaling_mode;
+            dst_width = dest_width;
+            dst_height = dest_height;
+        }
 
-        output_info.width(dest_width);
-        output_info.height(dest_height);
+        std::vector<unsigned> maximum_size;
+        if (max_size.size()) {
+            if(max_size.size() == 1) {
+                maximum_size = {max_size[0], max_size[0]};
+            } else if(max_size.size() == 2) {
+                maximum_size = {max_size[0], max_size[1]}; // {width, height}
+            } else {
+                THROW("The length of max_size vector exceeds the image dimension.")
+            }
+        }
+
+        // Determine the max width and height to be set to the output info
+        std::vector<unsigned> output_info_size(2, 0);
+        if (maximum_size.size() && maximum_size[0] != 0 && maximum_size[1] != 0) {
+            // If max_size is passed by the user, the resized images cannot exceed the max size,
+            output_info_size = maximum_size;
+        } else {
+            std::vector<unsigned> src_size = {input->info().width(), input->info().height_single()};
+            std::vector<unsigned> dst_size = {dst_width, dst_height};
+
+            // compute the output info width and height wrt the scaling modes and roi passed
+            get_max_resize_width_and_height(src_size, dst_size, resize_scaling_mode, output_info_size);
```

Comment:
There is only few lines of code inside the fun get_max_resize_width_and_height. It is better to replace that code here and get rid of the function